Book-A-Desk.Api.Tests/FilteredBookingsTest.fs

```diff
@@ -0,0 +1,90 @@
+module Book_A_Desk.Api.Tests.FilteredBookingsTest
+
+open FsToolkit.ErrorHandling
+open System
+open Xunit
+
+open Book_A_Desk.Api.Models
+open Book_A_Desk.Domain.Office.Domain
+open Book_A_Desk.Infrastructure.DynamoDbEventStore
+open Book_A_Desk.Domain
+open Book_A_Desk.Infrastructure
+
+open Book_A_Desk.Domain.Events
+open Book_A_Desk.Domain.Reservation
+open Book_A_Desk.Domain.Reservation.Queries
+
+open Book_A_Desk.Domain.Reservation.Events
+
+
+let officeId = Guid.NewGuid ()
+let date = DateTime(2100,02,01)
+
+let createDeskBookedEvent date officeId email :DomainEvent =
+    ({
+            Date = date
+            EmailAddress = "anEmail" |> EmailAddress
+            OfficeId = officeId |> OfficeId
+    }: Events.DeskBooked) |> DeskBooked |> ReservationEvent
+let bookings: DomainEvent seq = 
+    [
+        createDeskBookedEvent date officeId "lasnikr"
+        createDeskBookedEvent date officeId "user"
+        createDeskBookedEvent date officeId "dummy"
+    ]
+        
+
+let mockProvideEventStore =
+        {
+            GetEvents = fun _ -> bookings |> Ok |> async.Return
+            AppendEvents = fun _ -> failwith "should not be called"
+        } : DynamoDbEventStore.DynamoDbEventStore
+
+let getBookingEvents : Option<seq<DomainEvent>> =
+    async {
+        let! result = mockProvideEventStore.GetEvents()
+        match result with
+        | Ok events -> return Some events
+        | Error errorMessage -> return None
+    }
+    |> Async.RunSynchronously
+
+let getBookingsFrom (bookingEvents: Option<seq<DomainEvent>>) (date : DateTime option)  (officeId : OfficeId option) (email : EmailAddress option) =
```

Comment:
I'm not quite sure I understand what this code is doing. It seems to be a copy-paste from the BookingsHttpHandler. I suppose by doing this you are testing the code in the BookingsHttpHandler, but it means you would have to modify the test every time you modify the BookingsHttpHandler. This is quite bad because it is easy for someone to forget and all of a sudden, you are no longer running valid tests.

To fix this, you need to call the TestServer with the API call directly. You can see in GetBookingsTests.fs how to do it, but don't hesitate to ask for clarification if needed.